instagram-page-react/src/pages/profilePage/reducers/selectors/profilePage.posts.js

```diff
@@ -1,16 +1,30 @@
 import { compose } from 'recompose';
+import { defaultMemoize } from 'reselect';
 
 // lodash
 import _property from 'lodash/property';
 
+// utils
+import reverseArray from "../../../../utils/reverseArray";
+
 const getProfilePageState = _property('profilePage');
 
 const getPostListState = compose(
     _property('posts'),
     getProfilePageState
 );
 
-export const getPosts = compose(
+const getPostsList = compose(
     _property('data'),
     getPostListState
 );
+
+const reversePosts = defaultMemoize(reverseArray);
+
+export const getPosts = (state, arePostsReversed = false) => {
```

Comment:
We can avoid adding this flag here, we can leverage inversion of control here
Since the component is aware of what array it wants, we should make two selectors and let the component decide which one to call

Please refer to [this](https://kentcdodds.com/blog/inversion-of-control) for understanding inversion of control